server/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -171,8 +174,11 @@ private void setup(boolean addShutdownHook, Environment environment) throws Boot
 
         final long cacheSize = SnapshotsService.SNAPSHOT_CACHE_SIZE_SETTING.get(settings).getBytes();
         final long regionSize = SnapshotsService.SNAPSHOT_CACHE_REGION_SIZE_SETTING.get(settings).getBytes();
-        final int numRegions = Math.toIntExact(cacheSize / regionSize);
-        final long fileSize = numRegions * regionSize;
+        final long smallRegionSize = Math.min(SNAPSHOT_CACHE_SMALL_REGION_SIZE.get(settings).getBytes(), regionSize / 2);
```

Comment:
This should obviously be deduplicated with the same logic in the frozen cache class. I am not 100% on this being the right set of settings though so I didn't do anything about that in this PR. It seems like we need to either separately define the size of small and large page caches or a ratio between both. I went with the ratio because that's simpler but it still seems somewhat unwieldy.